dashboard/app/controllers/api/v1/pd/applications_controller.rb

```diff
@@ -187,14 +190,21 @@ def update
 
     # DELETE /api/v1/pd/applications/1
     def destroy
+      return head :forbidden if @application.status == 'incomplete' && !current_user.workshop_admin?
       @application.destroy
     end
 
     # GET /api/v1/pd/applications/search
     def search
       email = params[:email]
       user = User.find_by_email email
-      filtered_applications = @applications.where(
+
+      # only workshop admins can see incomplete applications
+      filtered_applications = current_user.workshop_admin? ? @applications.where(
```

Comment:
You might be able to use CanCan here too. You could filter out the applications that the user cannot read and then filter to the applications for the type and type. I don't know how that would the efficiency of this query.

Reply:
I couldn't get this to work ... It looks like it hides all of them rather than only some. When I talked to Dave about it on Wednesday, he mentioned CanCan is best for denying access to endpoints, not for the logic in between ... so we might be stuck with this. But it's nice we can use it for the :show, :destroy, and :update endpoints!

Reply:
Thanks for trying

Reply:
Good thought Dani. the way Meg has it looks best to me, both for the reasons that Meg mentioned and the efficiency issue that Dani mentioned.